src/control/server/storage/bdev/backend_test.go

```diff
@@ -1329,6 +1437,22 @@ func TestBackend_Prepare(t *testing.T) {
 				}
 			}
 			test.AssertEqual(t, tc.expHpCleanCall, hpCleanCall, "unexpected clean hugepages call")
+
+			entries, err := os.ReadDir(testDir)
+			if err != nil {
+				t.Fatal(err)
+			}
+			var locksRemaining []string
+			for _, v := range entries {
+				if v.IsDir() {
+					continue
+				}
+				locksRemaining = append(locksRemaining, v.Name())
+			}
+
+			if diff := cmp.Diff(tc.expLocksRemaining, locksRemaining); diff != "" {
+				t.Fatalf("\nunexpected list of lockfiles remove (-want, +got):\n%s\n", diff)
+			}
```

Comment:
Huh, so this test is creating real lockfiles? Do the mocks touch the real filesystem in this case?

Reply:
yes, tests create the files in a test directory and then verifying the expected locks get removed during the prepare call

Reply:
To be honest, with this approach it doesn't make much sense to use a mock for `Clean` at all, if it re-implements most/all of the functionality of the real method.

I argue again for splitting up unit testing along these interface boundaries. If we want to verify the real `NvmeImpl` deletes the lock files it is asked to delete, we should test `NvmeImpl.Clean()` with various inputs against real files. To see if `MockNvmeImpl.Clean()` was called with the correct parameters, you could have it save off something about its inputs before returning.